inst/scripts/make-grieneisen-ts-data.R

```diff
@@ -0,0 +1,111 @@
+library(SummarizedExperiment)
+library(TreeSummarizedExperiment)
+library(dplyr)
+library(phyloseq)
+
+#Sample phenotype data
+samples <- read.csv("git_metadata_and_community_phenotypes.csv")
+rownames(samples) <- samples$sample
+samples$sample <- NULL
+
+#Abundance table
+counts <- readRDS("git_ASV_table.RDS")
+
+#sequence information
+ASV_Sequence <- counts$asv_sequence
+ref_seq <- data.frame(ASV_Sequence)
+rownames(ref_seq) <- counts$asv_id
+
+#Taxonomic mapping table
+tax_1 <- select(counts, domain, phylum, class, order, family, genus, asv_id)
+rownames(tax_1) <- counts$asv_id
+
+# In order to make sure that the samples match between the abundance and phenodata tables
+counts_trimmed <- counts[, rownames(samples)]
+
+#Phylogenetic tree
+pseq <- readRDS("asv_10prev_phyloseq_for_lahti.RDS")
+tree <- phy_tree(pseq)
+
+#converting csv file into rds file
+saveRDS(samples, "samples.rds")
+
+#Metadata Table
+samples_1 <- samples %>% dplyr::rename(c(
+  Number = "X",
+  Baboon = "baboon_id",
+  Date = "collection_date",
+  Sex = "sex",
+  Age = "age",
+  Group = "social_group" ,
+  "Group_Size" = "group_size",
+  "Rain_Month_mm" = "rain_month_mm",
+  Season = "season",
+  Month = "month",
+  Readcount = "readcount",
+  Plate = "plate",
+  "Hydro_Year" = "hydro_year",
+  "Post_PCR_DNA_ng" = "post_pcr_dna_ng",
+  "Diet_Shannon_h" = "diet_shannon_h",
+  "ASV_Richness" = "asv_richness",
+  "ASV_Shannon_h" = "asv_shannon_h",
+  PC1_bc = "pc1_bc",
+  PC2_bc = "pc2_bc",
+  PC3_bc = "pc3_bc",
+  PC4_bc = "pc4_bc",
+  PC5_bc = "pc5_bc",
+  ))
+
+samples <- data.frame(samples_1)
+
+#Taxonomy Table
+saveRDS(tax_1, "tax.rds")
+tax <- tax_1  %>% dplyr::rename(c(
+  Domain = "domain",
+  Phylum = "phylum",
+  Class = "class",
+  Order = "order",
+  Family = "family",
+  Genus = "genus",
+  ASV = "asv_id"
+  ))
+
+#Arranging counts
+rownames(counts) <- counts$asv_id
+counts[, c("asv_id", "asv_sequence", "domain", "phylum", "class", "order", "family", "genus")]  <- NULL
+
+#sequence information
+refSeq <- DNAStringSet(ASV_Sequence, start=NA, end=NA, width=NA, use.names=TRUE)
+
+tse <- TreeSummarizedExperiment(assays = list(counts = counts_trimmed),
+                                colData = samples_1,
+                                rowData = tax,
+                                referenceSeq = refSeq)
+
+#Subsetting
+tse <- tse[tree$tip.label, ]
+

```

Comment:
Do subsetting with this instead:

`tse <- ape::keep.tip(phy = tree, tip = rowLinks(tse[tree$tip.label,])$nodeNum)`

This will collapse the tree, meaning that the extra tree tips will become pruned. 

You can test both ways to see the difference.

Or if this does not work let us have another look.

Reply:
Is the subsetting necessary? Data is lost that way isn't it?

Reply:
The problem at the moment is that we do not have access to the full tree, or the full set of reference sequences for this data. We only have access to tree that covers ASVs with population prevalence >10%.

We have asked for the full tree but it seems that there may be notable delays due to certain things, and we would like to implement this with the data that is currently available. If we can get access to the full tree later, we can update the data then.

Reply:
Nice! I think you can maybe toy around with it some more. Instead of using the `rowTree` setter, you can just save the tree separatly.

Internally `changeTree` is used to add the tree

https://github.com/microbiome/microbiomeDataSets/blob/04452ec9fcb53ff30d4b5bf7693441145c8c0b24/R/utils.R#L224

The way `changeTree` is working, is that non matching rows get deleted, basically the same thing you are doing manually.

So if you upload the full count dataset, which will not change later one, you don't need to reupload it again, just the full tree and reference sequences need to be added later. With two separate loading functions, you can serve the full and the subsetted data set, but from the same count data.

edit: So this actually doesn't work, because you cannot have two tree for the same identifier... damn. Let me think about it.

Reply:
Nope, no alternative to uploading the subset and later the full dataset...